### PR TITLE
Vwif

### PR DESCRIPTION
**Pull request by crogers1**
Implement vwif support in xenmgr.

OXT-973

### CODE REVIEW

**Comment by jean-edouard** [comment]
Didn't work for me... After starting a VM and connecting the laptop to the wifi, the disconnect node is still at 1.

File: xenmgr/Vm/Monitor.hs [hunk 1]
```diff
@@ -134,15 +137,19 @@ submitVmEvent m e = liftIO $ (vmm_submit m) e
 
 insertDefaultEvents :: VmMonitor -> Rpc ()
 insertDefaultEvents m = let uuid = vmm_uuid m in do
+    isNdvm <- readConfigPropertyDef uuid vmType ""
```
Comment (jean-edouard) [inline xenmgr/Vm/Monitor.hs]: Looks like this value is more `vmType` than `isNdvm`, no?

File: xenmgr/Vm/Monitor.hs [hunk 2]
```diff
@@ -160,22 +167,33 @@ insertDefaultEvents m = let uuid = vmm_uuid m in do
           _ -> return ()
     whenPowerState _ = return ()
 
+    whenNetState state backend = 
+        case state of
+            30  -> manageFrontVifs False (ND.netbackToUuid backend) --Network backend disconnected, handle it...
+            100 -> manageFrontVifs True (ND.netbackToUuid backend)
```
Comment (jean-edouard) [inline xenmgr/Vm/Monitor.hs]: What are these numbers? Are they documented somewhere?

Comment (crogers1) [inline xenmgr/Vm/Monitor.hs]: They're integers network manager uses to enumerate device state.  The UI was already hooked into these state updates, so it has them defined as constants in https://github.com/OpenXT/toolstack-data/blob/master/dist/script/constants.js#L205.

I'd be fine with macro-ing them to something descriptive.

File: xenmgr/Vm/Monitor.hs (hunk 1 above)
Comment (crogers1) [inline xenmgr/Vm/Monitor.hs]: I suppose so, I can change it so it's more readable.

**Comment by richpersaud** [comment]
Please include JIRA ticket number.

**Comment by rossphilipson** [comment]
Tested, working. Merging soon.

**Pull request merged**